spec/factories/miq_server.rb

```diff
@@ -1,18 +1,34 @@
 FactoryGirl.define do
   factory :miq_server do
+    transient do
+      # reference to the miq_server (typically the master) that knows the zone
+      master nil
+    end
     sequence(:name) { |n| "miq_server_#{seq_padded_for_sorting(n)}" }
     last_heartbeat  { Time.now.utc }
     status          "started"
     started_on      { Time.now.utc }
     stopped_on      ""
     version         '9.9.9.9'
+
+    trait :my_server do
+      guid           { MiqUUID.new_guid }
+      zone           { FactoryGirl.build(:zone) }
+      after(:create) do |ms|
+        MiqServer.stub(:my_guid).and_return(ms.guid)
+        MiqServer.my_server_clear_cache
+      end
+    end
+
+    after(:build) do |ms, e|
+      ms.guid ||= MiqUUID.new_guid if e.master
+      ms.zone ||= e.master.zone if e.master.respond_to?(:zone)
+    end
   end
 
   factory :miq_server_master, :parent => :miq_server do
+    guid           { MiqUUID.new_guid }
```

Comment:
Should this just move up to the base miq_server?  All server would have unique guids.